src/pqTestConnector/PqServiceHostClientLight.ts

```diff
@@ -0,0 +1,587 @@
+    constructor() {
+        super("Cannot connect to the pqServiceHost");
+    }
+}
+
+export class PqInternalError extends Error {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    constructor(message: string, public readonly data: any) {
+        super(message);
+    }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function getInternalErrorMessage(innerError: any): string {
+    if (typeof innerError === "string") {
+        return innerError;
+    } else if (typeof innerError === "object") {
+        if (typeof innerError["Message"] === "string") {
+            return innerError["Message"];
+        } else if (typeof innerError["Details"] === "string") {
+            return innerError["Details"];
+        } else if (typeof innerError["message"] === "string") {
+            return innerError["message"];
+        } else if (typeof innerError["details"] === "string") {
+            return innerError["details"];
+        }
+    }
+
+    return JSON.stringify(innerError);
+}
+
+type OmittedPqTestMethods = "currentExtensionInfos" | "currentCredentials" | "ForceShutdown" | "Ping";
+
+// eslint-disable-next-line @typescript-eslint/typedef
+export const INIT = "PqServiceHostClientEvent_Init" as const;
+// eslint-disable-next-line @typescript-eslint/typedef
+export const RETRYING = "PqServiceHostClientEvent_Retrying" as const;
+// eslint-disable-next-line @typescript-eslint/typedef
+export const DISCONNECTED = "PqServiceHostClientEvent_Disconnected" as const;
+// eslint-disable-next-line @typescript-eslint/typedef
+export const READY = "PqServiceHostClientEvent_Ready" as const;
+// eslint-disable-next-line @typescript-eslint/typedef
+export const DISPOSED = "PqServiceHostClientEvent_Disposed" as const;
+
+export class PqServiceHostClientLight
+    extends EventEmitter
+    implements Omit<IPQTestService, OmittedPqTestMethods>, IDisposable
+{
+    public static readonly ExecutableName: string = "PQServiceHost.exe";
+    public static readonly ExecutablePidLockFileName: string = "PQServiceHost.pid";
+    public static readonly ExecutablePortLockFileName: string = "PQServiceHost.port";
+
+    pqTestReady: boolean = false;
+    pqTestLocation: string = "";
+    pqTestFullPath: string = "";
+
+    private firstTimeStarted: boolean = true;
+    protected readonly sessionId: string = vscode.env.sessionId;
+    protected jsonRpcSocketClient: JsonRpcSocketClient | undefined = undefined;
+    // private pingTimer: NodeJS.Timer | undefined = undefined;
+    protected lastPqRelatedFileTouchedDate: Date = new Date(0);
+    protected _disposables: Array<IDisposable> = [];
+
+    public get pqServiceHostConnected(): boolean {
+        return this.jsonRpcSocketClient?.status === OPEN;
+    }
+
+    constructor(protected readonly outputChannel: PqSdkOutputChannelLight) {
+        super();
+    }
+
+    protected onPostConnected(): void {
+        // noop
+    }
+
+    protected onPreDisConnected(): void {
```

Comment:
minor nit:  `onPreDisconnected` don't capitalize the C.

Reply:
I've seen other APIs refer to the pre/post as 

pre: OnDisconnecting
post: OnDisconnected

pre: OnConnecting
post: OnConnected

but I don't feel strongly about the naming.

Reply:
rename onPostConnected into onConnected;
rename onPreDisConnected into onDisconnecting;
rename onPreRestartinginto onReconnecting;